cpp/src/arrow/compute/kernels/codegen_internal_test.cc

```diff
@@ -202,6 +202,8 @@ TEST(TestDispatchBest, CommonTemporalResolution) {
   ASSERT_EQ(TimeUnit::MILLI, CommonTemporalResolution(args.data(), args.size()));
   args = {time64(TimeUnit::MICRO), duration(TimeUnit::NANO)};
   ASSERT_EQ(TimeUnit::NANO, CommonTemporalResolution(args.data(), args.size()));
+  args = {duration(TimeUnit::SECOND), int64()};
+  ASSERT_EQ(TimeUnit::SECOND, CommonTemporalResolution(args.data(), args.size()));
```

Comment:
This looks unexpected to me since `int64` is not a temporal at all. @lidavidm What do you think?

Reply:
I think this is fine, but maybe we need a better name for what this function does. 

That said, I don't think this change is needed for this function to work, anyways. There's only a single temporal argument so there isn't really a meaningful common resolution to cast to.

Reply:
To be honest, the function doesn't seem to have changed :-). It's just the additional test that surprised me (but adding a test for existing behaviour is useful, thank you @rok ).

Reply:
Well duration has a unit and int64 is dimensionless. The goal of the function is to find the best common unit and I'd argue it still does.